vars/buildPlugin.groovy

```diff
@@ -134,12 +134,23 @@ def call(Map params = [:]) {
                         final String availableProxyProviders = configuredAvailableProxyProviders.split(',')
                         // Configure Maven settings if the requested provider is valid and available
                         if (validProxyProviders.contains(requestedProxyProvider) && availableProxyProviders.contains(requestedProxyProvider)) {
-                          echo "INFO: using artifact caching proxy from '${requestedProxyProvider}' provider."
-                          usingArtifactCachingProxy = true
-                          configFileProvider(
-                              [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", variable: 'MAVEN_SETTINGS')]) {
-                                infra.runMaven(mavenOptions, jdk, null, env.MAVEN_SETTINGS, addToolEnv)
-                              }
+                          boolean healthCheckOK = false
+                          final String healthCheckScript = "curl --fail https://repo.${requestedProxyProvider}.jenkins.io/healthz"
```

Comment:
Note that this will print a warning because you are interpolating a Groovy variable in a `sh` arg.

Reply:
Should I instead set an env var with the provider URL to use it latter with `sh` and `bat`?